includes/connect/class-wc-stripe-connect-api.php

```diff
@@ -55,7 +56,8 @@ public function get_stripe_oauth_init( $return_url ) {
 				'businessData' => $business_data,
```

Comment:
We need to read the account info for the correct `mode`, and include the `accountId` in the payload only if `account.controller.type` === `application` 
- https://docs.stripe.com/api/accounts/object#account_object-controller-type

Reply:
I assume this is mostly for the re-auth flow right? ie if they are already connected and their account has an `application` type controller, we need to pass the account ID to api.woocommerce.com so it can generate the correct type of oauth URL.

Reply:
Correct, the connect server checks the account compatibility, and generates the correct OAuth URL to be used for the re-connect flow.

During the initial onboarding (new installs), there is no account_id, so we default to the Connect OAuth flow.

Reply:
Ok sounds good. I've made the necessary changes in [6acaf6d](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3320/commits/6acaf6de50ad91480f62679b1779a648d0b801de)